js/common/lib/tensor-impl.ts

```diff
@@ -168,6 +168,8 @@ export class Tensor implements TensorInterface {
             }
           } else if (arg1 instanceof typedArrayConstructor) {
             data = arg1;
+          } else if (isFloat16Array !== undefined && isFloat16Array(arg1)) {
+            data = arg1 as InstanceType<typeof Float16Array>;
```

Comment:
does the code actually call into this if branch? I assume that map NUMERIC_TENSOR_TYPE_TO_TYPEDARRAY_MAP should already set with key 'float16', so it should be handled in condition (arg1 instanceof typedArrayConstructor)